MediaPicker/src/main/java/org/wordpress/android/util/WPMediaUtils.java

```diff
@@ -0,0 +1,218 @@
+        String state = Environment.getExternalStorageState();
+        if (!state.equals(Environment.MEDIA_MOUNTED)) {
+            showSDCardRequiredDialog(context);
+            return null;
+        } else {
+            try {
+                return getLaunchCameraIntent(context, applicationId, callback);
+            } catch (IOException e) {
+                // No need to write log here
+                return null;
+            }
+        }
+    }
+
+    private static Intent getLaunchCameraIntent(Context context, String applicationId, LaunchCameraCallback callback)
+            throws IOException {
+        File externalStoragePublicDirectory = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DCIM);
+        String mediaCapturePath =
+                externalStoragePublicDirectory + File.separator + "Camera" + File.separator + "wp-" + System
+                        .currentTimeMillis() + ".jpg";
+
+        // make sure the directory we plan to store the recording in exists
+        File directory = new File(mediaCapturePath).getParentFile();
+        if (directory == null || (!directory.exists() && !directory.mkdirs())) {
+            try {
+                throw new IOException("Path to file could not be created: " + mediaCapturePath);
+            } catch (IOException e) {
+                Log.e(TAG, "", e);
+                throw e;
+            }
+        }
+
+        Uri fileUri;
+        try {
+            fileUri = FileProvider.getUriForFile(context, applicationId + ".provider", new File(mediaCapturePath));
+        } catch (IllegalArgumentException e) {
+            Log.e(TAG, "Cannot access the file planned to store the new media", e);
+            throw new IOException("Cannot access the file planned to store the new media");
+        } catch (NullPointerException e) {
+            Log.e(TAG, "Cannot access the file planned to store the new media - "
+                    + "FileProvider.getUriForFile cannot find a valid provider for the authority: "
+                    + applicationId + ".provider", e);
+            throw new IOException("Cannot access the file planned to store the new media");
+        }
+
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, fileUri);
+
+        if (callback != null) {
+            callback.onMediaCapturePathReady(mediaCapturePath);
+        }
+        return intent;
+    }
+
+    public static int getPlaceholder(String url) {
+        if (MediaUtils.isValidImage(url)) {
+            return R.drawable.ic_image_white_24dp;
+        } else if (MediaUtils.isDocument(url)) {
+            return R.drawable.ic_pages_white_24dp;
+        } else if (MediaUtils.isPowerpoint(url)) {
+            return R.drawable.media_powerpoint;
+        } else if (MediaUtils.isSpreadsheet(url)) {
+            return R.drawable.media_spreadsheet;
```

Comment:
Related issue: [Convert all png icons to svg (#26)](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/issues/26)